### ISSUE TITLE
[request]Guided scale and orientation

### ISSUE THREAD

**Issue by pr0eXeler0** [alicevision/meshroom#923]
No matter how the images are taken you can be sure the model is always going to be rotated completely wrong and scale will be off. 

For geotagged images I've previously suggested it should be possible to use that date in order to establish a more realistic scale and better orientation.
But for regular meshes that are done from images from phones or cameras with no geo tagging, it would be helpful if one could guide the setup in Meshroom.

If I have an image of a wall, I could put markers at two places and enter a value for distance between these points..

Also, and this one I dont really know if it would work but if one could reorient the model by guiding the software to align to ground..

**Comment by natowi** [comment]
> model is always going to be rotated 

You can use the SfMTransform and select FromSingleCamera to use the orientation from one camera as origin, including orientation.

> put markers at two places and enter a value for distance between these points..

This will be possible with the next release using CCTAGs
(alicevision/AliceVision#695 & #652)

**Comment by skinkie** [comment]
@natowi your comments are always very helpfull :-)

Is FromSingleCamera also able to pick up Exif data here?

**Comment by natowi** [comment]
> Is FromSingleCamera also able to pick up Exif data here?

No, it will just set the camera to be set as origin. But you can combine two SfMTransform nodes to set one camera you know is aligned correctly (horizontal, frontal capture of a statue for example) as origin [fromsinglecamera] and use the second node [transformation] to adjust the orientation (from panorama to portrait view for example). Keep the additional scale at 1 to better compare the changes to the original sfm.

**Comment by skinkie** [comment]
@natowi do you think it is a good idea to have an Exif option here? Or should it not be added here, but instead directly at camera init?

**Comment by natowi** [comment]
@skinkie I think this is one of the many features waiting to be implemented. It should be addressed in the CameraInit node.

**Comment by endolith** [comment]
Ideally it would pick up the camera orientation from the EXIF data to orient the model

```
User comment                     : Yaw:140.11209,Pitch:-5.279445455007514,Roll:3.6983407727132422 
…
Orientation                      : right-top (6)
```
